.github/workflows/pull-request-ci.yaml

```
@@ -199,14 +199,22 @@ jobs:
           echo "::set-env name=SSH_AUTH_SOCK::${SSH_AUTH_SOCK}"
           echo "::set-env name=SSH_AGENT_PID::${SSH_AGENT_PID}"
 
-      # set GKE secret. TBD: use vault instead
+      - name: Query vault for GKE secrets
+        id: vault
+        uses: hashicorp/vault-action@v2.0.1
+        with:
+          exportEnv: false
+          url: https://volt.cap.explore.suse.dev/
+          token: ${{ secrets.VAULT_TOKEN }}
+          secrets: |
+            secret/data/cloud-providers gke.GKE_CRED_JSON | GKE_CRED_JSON
+
+      # set GKE secret.
       - name: set GKE_CRED_JSON
         run: |
           json_file="$(mktemp)"
-          echo "$GKE_CRED_JSON" > "${json_file}"
+          echo "${{ steps.vault.outputs.GKE_CRED_JSON }}" > "${json_file}"
           echo "::set-env name=GKE_CRED_JSON::${json_file}"
-        env:
-          GKE_CRED_JSON: ${{ secrets.GKE_CRED_JSON }}
 
```

Here we are not setting the env var `GKE_CRED_JSON` with  `GKE_CRED_JSON: ${{ secrets.GKE_CRED_JSON }}` anymore.

I think we need to set `GKE_CRED_JSON` to the path of `"${json_file}"`, as Catapult expects an absolute path on it, pointing to a file containing the gke json secret.

Something like:
```suggestion
        env:
          GKE_CRED_JSON: $(readlink -f ${json_file})

```

---

Will fix.

---

PR rebased, conflicts resolved. Now I can do the fix here.

---

After resolving conflicts the configuration looks like this:
```
        run: |
          json_file="$(mktemp)"
          echo "${{ steps.vault.outputs.GKE_CRED_JSON }}" > "${json_file}"
          echo "GKE_CRED_JSON=${json_file}" >> "${GITHUB_ENV}"
```

This is the new way of setting the GHA environment. Note that `json_file` is a path generated by `mktemp`.
That is already an absolute path.
I do not see anything to be done here anymore.